source/plugins/ruby/kubelet_utils.rb

```diff
@@ -56,74 +59,68 @@ def get_node_allocatable(cpu_capacity, memory_capacity)
         begin
           systemReserved_cpu = parsed_response["kubeletconfig"]["systemReserved"]["cpu"]
           if systemReserved_cpu.nil? || systemReserved_cpu == ""
-            systemReserved_cpu = "0"
+            systemReserved_cpu = "0.0"
           end
           @log.info "get_node_allocatable::systemReserved_cpu  #{systemReserved_cpu}"
         rescue => errorStr
           # this will likely always reach this condition for AKS ~ only applicable for hyrid + MDM combination
           @log.error "Error in get_node_allocatable::systemReserved_cpu: #{errorStr}"
-          systemReserved_cpu = "0"
-          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+          systemReserved_cpu = "0.0"
         end 
 
         begin
           explicitlyReserved_cpu = parsed_response["kubeletconfig"]["reservedCPUs"]
           if explicitlyReserved_cpu.nil? || explicitlyReserved_cpu == ""
-            explicitlyReserved_cpu = "0"
+            explicitlyReserved_cpu = "0.0"
           end
           @log.info "get_node_allocatable::explicitlyReserved_cpu  #{explicitlyReserved_cpu}"
         rescue => errorStr
           # this will likely always reach this condition for AKS ~ only applicable for hyrid + MDM combination
           @log.error "Error in get_node_allocatable::explicitlyReserved_cpu: #{errorStr}"
-          explicitlyReserved_cpu = "0"
-          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::explicitlyReserved_cpu: #{errorStr}")
+          explicitlyReserved_cpu = "0.0"
         end 
 
         begin
            systemReserved_memory = parsed_response["kubeletconfig"]["systemReserved"]["memory"]
            if systemReserved_memory.nil? || systemReserved_memory == ""
-            systemReserved_memory = "0"
+            systemReserved_memory = "0.0"
            end
            @log.info "get_node_allocatable::systemReserved_memory #{systemReserved_memory}"
         rescue => errorStr
            @log.error "Error in get_node_allocatable::systemReserved_memory: #{errorStr}"
-           systemReserved_memory = "0"
-           ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+           systemReserved_memory = "0.0"
         end 
 
         begin
           evictionHard_memory = parsed_response["kubeletconfig"]["evictionHard"]["memory.available"]
           if evictionHard_memory.nil? || evictionHard_memory == ""
-            evictionHard_memory = "0"
+            evictionHard_memory = "0.0"
           end
           @log.info "get_node_allocatable::evictionHard_memory #{evictionHard_memory}"
         rescue => errorStr
           @log.error "Error in get_node_allocatable::evictionHard_memory: #{errorStr}"
-          evictionHard_memory = "0"
-          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
```

Comment:
why remove the exception telemetry?

Reply:
I feel like this is logging too much telemetry in our appinsights resource. Right now its being logged in the filter_cadvisor2mdm.log file maintained on the node via the @log.error line and there is no action item to take from this telemetry point as I'm setting the default values to 0.0 if they don't exist in the response. 

I can add it back in again if you think its adding valuable information.

Reply:
I think it's good to keep. Since the input data comes from CAdvisor (which we don't control) this might alert us to an actual problem.

Reply:
I've added it back in again, thanks. Resolving this comment now.